performance-tests/src/main/java/io/kroxylicious/InvokerBenchmark.java

```diff
@@ -0,0 +1,174 @@
+        invoke(blackhole, state.arrayInvokerFilter2HasTwoInterfaces);
+    }
+
+    @Benchmark
+    public void testArrayInvokerFilter2HasFourInterfaces(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.arrayInvokerFilter2HasFourInterfaces);
+    }
+
+    @Benchmark
+    public void testArrayInvokerFilter2HasEightInterfaces(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.arrayInvokerFilter2HasEightInterfaces);
+    }
+
+    @Benchmark
+    public void testMapInvokerFilterHasOneInterface(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.mapInvokerFilterHasOneInterface);
+    }
+
+    @Benchmark
+    public void testMapInvokerFilterHasTwoInterfaces(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.mapInvokerFilterHasTwoInterfaces);
+    }
+
+    @Benchmark
+    public void testMapInvokerFilterHasFourInterfaces(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.mapInvokerFilterHasFourInterfaces);
+    }
+
+    @Benchmark
+    public void testMapInvokerFilterHasEightInterfaces(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.mapInvokerFilterHasEightInterfaces);
+    }
+
+    @Benchmark
+    public void testMapInvoker2FilterHasOneInterface(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.mapInvoker2FilterHasOneInterface);
+    }
+
+    @Benchmark
+    public void testMapInvoker2FilterHasTwoInterfaces(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.mapInvoker2FilterHasTwoInterfaces);
+    }
+
+    @Benchmark
+    public void testMapInvoker2FilterHasFourInterfaces(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.mapInvoker2FilterHasFourInterfaces);
+    }
+
+    @Benchmark
+    public void testMapInvoker2FilterHasEightInterfaces(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.mapInvoker2FilterHasEightInterfaces);
+    }
+
+    @Benchmark
+    public void testFieldInvokerFilterHasOneInterface(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.fieldInvokerFilterHasOneInterface);
+    }
+
+    @Benchmark
+    public void testFieldInvokerFilterHasTwoInterfaces(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.fieldInvokerFilterHasTwoInterfaces);
+    }
+
+    @Benchmark
+    public void testFieldInvokerFilterHasFourInterfaces(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.fieldInvokerFilterHasFourInterfaces);
+    }
+
+    @Benchmark
+    public void testFieldInvokerFilterHasEightInterfaces(BenchState state, Blackhole blackhole) {
+        invoke(blackhole, state.fieldInvokerFilterHasEightInterfaces);
+    }
+
+    private static void invoke(Blackhole blackhole, FilterInvoker filter) {
+        blackhole.consume(filter.shouldHandleRequest(ApiKeys.PRODUCE, ApiKeys.PRODUCE.latestVersion()));
```

Comment:
Is it worth including some non-handled requests in the benchmark. Given the discussions around sparely populated maps/arrays and fully populated ones.

Reply:
The Fetch request and response are non-handled. The test filters are:

```
OneInterfaceFilter implements ProduceResponseFilter

TwoInterfaceFilter implements ProduceResponseFilter, ProduceRequestFilter

FourInterfaceFilter implements ProduceResponseFilter, ProduceRequestFilter, CreateTopicsRequestFilter, CreateTopicsResponseFilter

EightInterfaceFilter implements ProduceResponseFilter, ProduceRequestFilter, CreateTopicsRequestFilter, CreateTopicsResponseFilter, DeleteTopicsRequestFilter, DeleteTopicsResponseFilter, DescribeGroupsRequestFilter, DescribeGroupsResponseFilter
```

Reply:
Ah nice!